config/routes.rb

```diff
@@ -1,3 +1,7 @@
 Rails.application.routes.draw do
   root 'pages#home'
+
+  match 'auth/:provider/callback', to: 'sessions#create', via: [:get, :post]
+  match 'auth/failure', to: redirect('/'), via: [:get, :post]
+  match 'signout', to: 'sessions#destroy', as: 'signout', via: [:get, :post]
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.